### ISSUE TITLE
[RLlib] [Bug] Despite providing correct config, function returns incorrect actions

### ISSUE THREAD

**Issue by tlangerak** [ray-project/ray#21417]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

In normalized action spaces the `agent.compute_single_action(obs)` does not use the `config[normalize_actions] = True` flag. However, the documentation suggests this. It only considers `agent.compute_single_action(obs, unsquash_action=True)`. Hence, despite providing a correct config file, the function returns actions that does not make sense. 

### Versions / Dependencies

- Ray 1.9.1 (Also tested on 1.8.0 and 2.0.0dev0)
- Windows 10, also tested on a cluster running Ubuntu 20
- Python 3.8
- both in torch and tf2 backends.

### Reproduction script
```
import ray
from ray.rllib.agents.ppo import PPOTrainer, DEFAULT_CONFIG
import gym
from gym import spaces
import numpy as np
import ray.rllib.agents.ppo as ppo
from ray.tune.registry import register_env

select_env = "IssueEnv"
rllib_config = DEFAULT_CONFIG.copy()
rllib_config["num_workers"] = 0
rllib_config['env'] = select_env
rllib_config["framework"] = "torch"
rllib_config["num_gpus"] = 0
rllib_config['seed'] = 123
rllib_config['normalize_actions'] = True  # <-- set normalize actions to True
rllib_config['evaluation_config'] = {
    "normalize_actions": True,
    "explore": False}  # <-- set normalize actions to True, just to be sure also in eval Config


class IssueEnv(gym.Env):
    metadata = {'render.modes': ['human']}

    def __init__(self):
        super().__init__()
        self.action_space = spaces.Box(low=np.array([0., ]),
                                       high=np.array([1.0]), dtype=np.float64)

        self._observation_space = {"state": spaces.Box(low=np.array([-1.]),
                                                       high=np.array([1.]), dtype=np.float64)}

        self.observation_space = spaces.Dict(self._observation_space)
        self.done = False
        self.add = {}  # whatever additional information there is
        self.reward = 0
        self.state = {'state': 0.}

    def reset(self):
        self.done = False
        self.add = {}  # whatever additional information there is
        self.reward = 0
        self.state['state'] = np.asarray([0.])
        return self.state

    def step(self, action):
        self.state['state'] = action[0]
        self.reward = np.sin(self.state['state'] * np.pi * 2)
        self.done = True
        return self.state, self.reward, self.done, {}

    def render(self):
        pass


ray.init()
register_env(select_env, lambda c: IssueEnv())
trainer = ppo.PPOTrainer(
    config=rllib_config,
    env=select_env,
)

# Obviously it wont converge, but it shows what happens
trainer.train()

trainer.save("/tmp/rllib_checkpoint")
eval_agent = PPOTrainer(rllib_config)
eval_agent.restore("/tmp/rllib_checkpoint/checkpoint_000001/checkpoint-1")
env = IssueEnv()
state = env.reset()
print(f"Default action: {eval_agent.compute_single_action(state)}")
state = env.reset()
print(f"Expected action: {eval_agent.compute_single_action(state, unsquash_action=True)}")
batch = eval_agent.workers.local_worker().sample()

for i in range(5):
    # since the state is directly the action, we expect action == new_obs. We see that this is not the case.
    state = env.reset()
    print(f"action: {batch['actions'][i]}, old_obs: {batch['obs'][i]}, new_obs/Expected action: {batch['new_obs'][i]}")

```

### Anything else

I also observed that agent.workers.local_worker().sample() returns unnormalized actions

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR! (If someone guides me through the process)

**Comment by sven1977** [comment]
I think this is solved by this PR here:
https://github.com/ray-project/ray/pull/21553

Similar issue:
https://github.com/ray-project/ray/issues/21552
